### PR TITLE
[spring] Add delete command for tanzu components

### PR DESCRIPTION
**Pull request by ninpan-ms**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### Related command
az spring application-configuration-service create/delete
az spring service-registry create/delete
az spring gateway create/delete
az spring api-portal create/delete

### General Guidelines

- [X] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [X] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repository and upgrade the version in the pull request but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
spring

File: src/spring/azext_spring/commands.py [hunk 1]
```diff
@@ -260,6 +260,8 @@ def load_command_table(self, _):
                               table_transformer=transform_service_registry_output)
         g.custom_command('bind', 'service_registry_bind')
         g.custom_command('unbind', 'service_registry_unbind')
+        g.custom_command('create', 'service_registry_create', table_transformer=transform_service_registry_output)
+        g.custom_command('delete', 'service_registry_delete')
```
Comment (zhoxing-ms) [inline src/spring/azext_spring/commands.py:264]: Do we need to consider adding `confirmation=True` for these delete commands?

**Comment by zhoxing-ms** [comment]
Please add the description of PR changes into the `HISTORY.rst`. In addition, if you want to release a new version for this PR, please upgrade the version defined in `setup.py`

**Comment by ninpan-ms** [comment]
> Please add the description of PR changes into the `HISTORY.rst`. In addition, if you want to release a new version for this PR, please upgrade the version defined in `setup.py`

Just added.

**Comment by zhoxing-ms** [comment]
Could you add some test cases for these new commands?

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**